frontend/appflowy_flutter/lib/plugins/database_view/application/cell/cell_data_persistence.dart

```diff
@@ -29,8 +29,8 @@ class TextCellDataPersistence implements CellDataPersistence<String> {
 @freezed
 class DateCellData with _$DateCellData {
   const factory DateCellData({
-    required DateTime date,
-    String? time,
+    required DateTime? dateTime,
+    required String? time,
```

Comment:
These are optional, but IIUC, at least one of them, shouldn't be null. We could consider adding an assertion to check that one is provided.

Also, consider exposing two factory constructors rather than one with two optional parameters. Here's an example.

```dart
class Example {
  factory Example.fromString({required String s}) => Example._(int.parse(s));
  factory Example.fromInt({required int i}) => Example._(i);
  // the main constructor here is private so that we don't have to have logic that deals with initializing 
  // the backing property, each constructor above does that for us.
  Example._(this.backingProperty);
  final int backingProperty;
}
```

Reply:
Again, these are fly-by comments. You can correct me if I'm wrong :)

Reply:
Done! Removed required fields if nullable